File: source_funcs/go_transcoder/SEARCH_INSERT_AND_DELETE_IN_A_SORTED_ARRAY_1/SEARCH_INSERT_AND_DELETE_IN_A_SORTED_ARRAY_1.go
package main

func f_gold(arr []int, n int, key int, capacity int) int {
	if n >= capacity {
		return n
	}
	var i int
	for i = n - 1; i >= 0 && arr[i] > key; i-- {
		arr[i+1] = arr[i]
	}
	arr[i+1] = key
	return n + 1
}

File: rust_funcs/cpp_transcoder/SEARCH_INSERT_AND_DELETE_IN_A_SORTED_ARRAY_1/SEARCH_INSERT_AND_DELETE_IN_A_SORTED_ARRAY_1.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn f_gold(arr: &mut [f32], n: u32, key: f32, capacity: u32) -> i32 {
    if n >= capacity {
        return n;
    }
    let mut i = n - 1;
    while i >= 0 && arr[i] > key {
        arr[i + 1] = arr[i];
        i -= 1;
    }
    arr[i + 1] = key;
    n + 1
}
